CMakeLists.txt

```diff
@@ -0,0 +1,294 @@
+cmake_minimum_required(VERSION 3.14 FATAL_ERROR)
+
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+project(bladebit)
+
+set(CMAKE_CONFIGURATION_TYPES release debug)
+
+if(NOT CMAKE_BUILD_TYPE)
+    set(CMAKE_BUILD_TYPE "release"
+        CACHE STRING "Possible values are: release, debug"
+        FORCE
+   )
+endif()
+message("Config: ${CMAKE_BUILD_TYPE}")
+
+
+# Grab BLS
+include(FetchContent)
+
+set(BUILD_BLS_PYTHON_BINDINGS "0" CACHE STRING "0")
+set(BUILD_BLS_TESTS "0" CACHE STRING "")
+set(BUILD_BLS_BENCHMARKS "0" CACHE STRING "")
+
+FetchContent_Declare(
+  bls
+  GIT_REPOSITORY https://github.com/Chia-Network/bls-signatures.git
+  GIT_TAG        1.0.6
+)
+FetchContent_MakeAvailable(bls)
+
+set(CMAKE_C_FLAGS_RELEASE_INIT)
+set(CMAKE_C_FLAGS_DEBUG_INIT)
+set(CMAKE_CXX_FLAGS_RELEASE_INIT)
+set(CMAKE_CXX_FLAGS_DEBUG_INIT)
+
+
+set(c_opts)
+set(link_opts)
+# set(cxx_opts)
+
+set(release_c_opts)
+set(debug_c_opts)
+set(dev_c_opts)
+
+set(release_link_opts)
+set(debug_link_opts)
```

Comment:
I wouldn't set these values to empty before appending to them below, considering they'll be set across all paths with no clashing possibility in this context, so just setting them below with meaningful values is enough.

Also, you don't seem to be using `dev_c_opts`?

Reply:
This is meant to be there for parity with my previous setup where I add/remove flags to the dev var while working/experimenting.  I declare them beforehand to be explicit about the present vars as it can get confusing once they fall into the if/elses.